README.md

```diff
@@ -61,4 +61,209 @@ The third option is relevant when you are planning to implement read-only
 inspection features.
 
 ## API
-Check `AmoClient` in `index.d.ts`
+
+Requests can be made by passing the releavant paramaters to `AmoClient`.
+
+You can check types in `index.d.ts`
+
+## Types
+```typescript
+// Uppercase Hex encoded address
+type HexEncodedAddress = string
+
+// Uppercase Hex encoded id
+type HexEncodedParcelID = string
+
+// DecimalString represents number as string
+// e.g 1000 -> "1000"
+type DecimalString = string
+```
+## AmoClient
+
+AmoClient uses [Axios](https://github.com/axios/axios#request-config) internally for making requests 
+
+```typescript
+import { AxiosRequestConfig } from "axios"
+
+class AmoClient {
+    constructor(
+        // Config for Tendermint REST API and ABCI query
+        nodeClient?: AxiosRequestConfig,
+        // Config for AMO storage API
+        storageClient?: AxiosRequestConfig,
+        // WebSocket URL for newBlock event
+        wsURL?: string
+    )
+
+```
+Return type of AmoClient method is `Promise<T>`
+
+## Tendermint REST API
+AMO blockchain is based on [Tendermint](https://github.com/tendermint/tendermint).
+We provide basic Tendermint [RPCs](https://docs.tendermint.com/master/rpc/).
+
+### fetchLastBlock()
+
+- Fetch latest block header
+
+### fetchBlock(height)
+
+- Fetch Block of specific height
+
+### fetchBlockHeaders(maxHeight, count)
+
+- Fetch Block headers 
+
+### fetchRecentBlockHeaders()
+
+- Fetch recent 20 block headers
+
+### fetchTx(hash)
+
+- Fetch transaction by hash
```

Comment:
We're formatting tx body in our own format. Explain it or add a reference to index.d.ts here.